iceoryx_utils/cmake/IceoryxPlatform.cmake

```diff
@@ -48,3 +48,32 @@ if(BUILD_STRICT)
         set(ICEORYX_WARNINGS ${ICEORYX_WARNINGS} -Werror)
     endif (  )
 endif(BUILD_STRICT)
+
+if(sanitize)
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+        # NOTE : This works only when iceoryx is built standalone , in which case CMAKE_SOURCE_DIR point to iceoryx_meta
+        set(ICEORYX_SANITIZER_BLACKLIST -fsanitize-blacklist=${CMAKE_SOURCE_DIR}/sanitizer_blacklist/asan_compile_time.txt)
+    endif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+    
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+        set(ICEORYX_SANITIZER_COMMON_FLAGS -fno-omit-frame-pointer -fno-optimize-sibling-calls -g -O1)
```

Comment:
is `-O1` required? I'm just asking because I'm not sure if it's overiden later on in cmake

Reply:
I think it wont break the sanitizer. Its more about getting a readable stack trace of the error.